src/crypto/sha256.cpp

```diff
@@ -623,6 +633,34 @@ std::string SHA256AutoDetect()
 #endif
 #endif
 
+#if defined(ENABLE_ARM_SHANI) && !defined(BUILD_BITCOIN_INTERNAL)
+    bool have_arm_shani = false;
+
+#ifdef __linux__
+#ifdef __arm__ // 32-bit
+    if (getauxval(AT_HWCAP2) & HWCAP2_SHA2) {
+        have_arm_shani = true;
+    }
+#endif
```

Comment:
I'm not sure if bitcoin core has a preference documented somewhere, but I prefer documenting what the endif is associated with such as
```suggestion
#endif // __arm__
```

Reply:
I checked all the other files in `src/crypto` and we use the endif comments only for the include guards.